Automattic/sensei
# Reports: Apply data filters to column total values

Fixes #4926

### Changes proposed in this Pull Request
- Totals were showing the same value agnostic of the filters like **Last Activity** or **Search**
- The changes were made on all three tables: **Courses**, **Students**, **Lessons**  

### Courses and how to test:
- Commit: https://github.com/Automattic/sensei/pull/5091/commits/7dd894396d8d9711618a87932af5d6cf3362b733
- Go to the Sensei LMS -> Reports
- Select the Courses tab
- Courses has 4 column totals: 
<img width="1330" alt="image" src="https://user-images.githubusercontent.com/7208249/166478774-7c16ab4c-13ec-4ebd-bfe1-cefae689fa06.png">
-  Make sure that the proper number is displayed in the totals when you: 
1. Don't have any filter
2. Have some Last Activity Filter
3. Search 

### Students and how to test:
- Commit:https://github.com/Automattic/sensei/pull/5091/commits/d2a3a67f66858d167769da5fc777ca3ffa678f61
- Go to the Sensei LMS -> Reports
- Select the Students tab
- Courses has 4 column totals: 
<img width="1336" alt="image" src="https://user-images.githubusercontent.com/7208249/166479219-18e6c3f1-93ed-4c5a-8d0e-aae5f285484b.png">
-  Make sure that the proper number is displayed in the totals when you: 
1. Don't have any filter
2. Have some Last Activity Filter
3. Search 


### Lessons and how to test:
- Commit: https://github.com/Automattic/sensei/pull/5091/commits/c97ee0191cb17b069e4a1ead70e7be7e942548c1
- Go to the Sensei LMS -> Reports
- Select the Lessons tab
- Select a course (or nothing is displayed)
- Courses has 6 column totals: 
<img width="1346" alt="image" src="https://user-images.githubusercontent.com/7208249/166479450-2d9cc6f3-4fa6-42f5-afc4-604d1c3c9357.png">
-  Make sure that the proper number is displayed in the totals when you: 
1. Don't have any filter
2. Have some Last Activity Filter
3. Search 

### Deprecated Methods: 
1. Sensei_Course::get_average_days_to_completion **replaced by** Sensei_Reports_Overview_List_Table_Courses::get_average_days_to_completion
